### PR TITLE
[KED-2210] Update npm dependencies

### PR DESCRIPTION
**Pull request by richardwestenra**
## Description

Update react-scripts and @babel/cli to new major versions, and update other dependencies to their most recent minor versions.

These updates are intended to force an update of chokidar, a subdependency of both.

See react-scripts v4 release notes: https://medium.com/frontend-digest/whats-new-in-create-react-app-4-809b24e4d6cb

I've also needed to make a bunch of minor changes to fix eslint errors, because of new eslint rules.

## Development notes

This is a first step towards fixing our CI (if this PR doesn't fix it). In future PRs I want to look into the following:
1. [ ] Fix the eslint error on the LazyList component (cc @bru5)
2. [ ] Update the rest of the dependencies to new major versions - note that [React 17 will depend upon a new enzyme adaptor, which has not yet been released](https://github.com/enzymejs/enzyme/issues/2429).
3. [ ] Pin the fsevents version in package.json's `optionalDependencies` field ([source](https://github.com/fsevents/fsevents/issues/301#issuecomment-572607085))

## QA notes

No expected behaviour changes.

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,26 +43,26 @@
     "d3-fetch": "^1.2.0",
     "d3-interpolate": "^1.4.0",
     "d3-interpolate-path": "^2.2.1",
-    "d3-scale": "^3.2.1",
+    "d3-scale": "^3.2.3",
```
Comment (studioswong) [inline package.json:46]: I wonder whether this minor update will have any effect on the flowchart? Might be worth a small regression test just to make sure.

Comment (richardwestenra) [reply package.json:46]: This PR doesn't introduce any new functionality, so if we felt it necessary to write some new tests for the flowchart component, shouldn't we do that in a different PR, as needing more tests is a property of that component, not of this PR? I would suggest creating a new JIRA issue for this issue but we've already closed two on this very subject.

Secondly, you and I looked into testing the D3 functionality in this component about a week ago, and we both agreed that it was difficult enough that we felt it wasn't worthwhile - hence the closed JIRA issues.

Thirdly, Liam is reportedly working on some new tests in this area.

Comment (bru5) [reply package.json:46]: Yeah I think if manual testing and the existing unit tests show no issues in this PR then it's fine, especially considering it's a patch update of the dependency. On the point of additional tests around this though, there's a bunch coming with the zoom refactor which should help double check this.

**Review by bru5 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (studioswong) [reply package.json:46]: Thanks both - for regression testing I meant 'manual QA' - sorry for the confusion!

**Review by studioswong — APPROVED** [review]
Thanks for the update @richardwestenra , I've done a bit of manual QA around the app to make sure the d3-zoom and react updates didn't break the app. 

One thing - I wonder for updates to more important packages that we use ( such as react and d3 features), do we normally engage a more thorough manual QA to ensure the app doesn't break? ( Asking about that given my previous teams would usually engage 2-3 people to perform the manual QA before merging package updates in).

**Comment by richardwestenra** [comment]
@studioswong We should probably discuss our QA practices as a team, but generally I'd say that the person who submits the PR should QA+review their own work, then the reviewer should also give it a QA. We don't have a full-time QA, and I don't think that we need multiple reviewers/QAs on every PR, but occasionally I do think a PR should get multiple sets of 👀 - especially a release PR. Does that sound ok to you?

**Pull request merged**